python/sdk/merlin/pyfunc.py

```diff
@@ -369,3 +393,209 @@ def infer(self, model_input: pandas.DataFrame) -> Union[numpy.ndarray,
+    protocol: Protocol = Protocol.HTTP_JSON,
+    debug: bool = False,
+):
+    """
+    Run pyfunc model locally using Docker. The function will log the model artifacts onto local mlflow, build a docker image, and run the container locally.
+
+    :param model_instance: instance of python function model
+    :param conda_env: path to conda env.yaml file
+    :param code_dir: additional code directory that will be loaded with ModelType.PYFUNC model
+    :param artifacts: dictionary of artifact that will be stored together with the model. This will be passed to PythonModel.initialize. Example: {"config": "config/staging.yaml"}
+    :param pyfunc_base_image: base image for building pyfunc model
+    :param port: port to expose the model
+    :param env_vars: dictionary of environment variables to be passed to the server
+    :param protocol: protocol to be used by the deployed model (default: HTTP_JSON)
+    :param debug: flag to enable debug mode that will print docker build log
+    """
+
+    model_name = str.lower(model_instance.__class__.__name__)
+
+    # Log model to local mlflow
+    print("Logging model to local MLflow")
+    tracking_uri = f"file:///tmp/merlin/pyfunc-models"
+    mlflow.set_tracking_uri(tracking_uri)
+    experiment = mlflow.set_experiment(model_name)
+
+    model_info = mlflow.pyfunc.log_model(
+        artifact_path="model",
+        python_model=model_instance,
+        conda_env=conda_env,
+        code_path=code_dir,
+        artifacts=artifacts,
+    )
+
+    context_path = experiment.artifact_location
+    if "file://" in context_path:
+        context_path = context_path.replace("file://", "")
+
+    dependencies_path = f"{context_path}/env.yaml"
+    shutil.copy(conda_env, dependencies_path)
+
+    artifact_path = f"{model_info.run_id}/artifacts/model"
+
+    run_pyfunc_local_server(
+        context_path=context_path,
+        dependencies_path=dependencies_path,
+        artifact_path=artifact_path,
+        model_name=model_name,
+        model_version="dev",
+        pyfunc_base_image=pyfunc_base_image,
+        port=port,
+        env_vars=env_vars,
+        protocol=protocol,
+        debug=debug,
+    )
+
+
+def run_pyfunc_local_server(
+    context_path: str,
+    dependencies_path: str,
+    artifact_path: str,
+    model_name: str,
+    model_version: str,
+    pyfunc_base_image: str = None,
+    port: int = 8080,
+    env_vars: Dict[str, str] = None,
+    protocol: Protocol = None,
+    debug: bool = False,
+):
+    if pyfunc_base_image is None:
+        if "dev" in VERSION or "0.0.0" in VERSION:
+            pyfunc_base_image = "ghcr.io/caraml-dev/merlin/merlin-pyfunc-base:0.38.1"
+        else:
+            pyfunc_base_image = (
+                f"ghcr.io/caraml-dev/merlin/merlin-pyfunc-base:{VERSION}"
+            )
+
+    dockerfile_path = copy_pyfunc_dockerfile(context_path)
+
+    _clone_merlin_repo(context_path)
+
+    image_tag = f"{model_name}-{model_version}"
```

Comment:
is this working for you? i ran a two example so far, they are not named nor tagged
![image](https://github.com/caraml-dev/merlin/assets/30390872/a8613bcd-f142-4467-9a2f-c2084991be01)